src/GNDStk/Node/src/detail.hpp

```diff
@@ -568,41 +507,88 @@ class child_ref<
                   childNodePtr.push_back(&(*c));
          }
       } catch (...) {
-         log::ctor("child_ref<...,void,many,...>" + detail::keyname(kwd));
+         log::ctor("child_ref<allow::many>" + detail::keyname(kwd));
          throw;
       }
    }
 
+   // ------------------------
+   // misc. functions
+   // ------------------------
+
    // size
    std::size_t size() const
    {
       return childNodePtr.size();
    }
 
    // operator[]
-   auto operator[](const std::size_t n) const
+   decltype(auto) operator[](const std::size_t n) const
```

Comment:
I got bitten by that one a couple of times myself :-)

Reply:
What I previously had actually worked, so there wasn't an error. For this PR, though, I consolidated some things, and the consolidation caused it to be necessary to change `auto` to `decltype(auto)`. Note that the function in question can return either a reference or a non-reference, depending on the `if constexpr`. The reference possibility is what necessitated the `decltype`.